### PR TITLE
Fix dispenser dying

### PR DESCRIPTION
**Pull request by MatusGuy**
- [x] make them unaffectable by rocks
- [x] fix cannon left & right sprites
- [x] unfreeze after a while

### CODE REVIEW

File: src/badguy/dispenser.cpp [hunk 1]
```diff
@@ -163,6 +163,12 @@ Dispenser::active_update(float dt_sec)
   }
 }
 
+void
+Dispenser::kill_fall()
+{
+  return;
```
Comment (tobbi) [inline src/badguy/dispenser.cpp:169]: Isn't the return kinda pointless? You can just leave the method empty.

Comment (MatusGuy) [reply src/badguy/dispenser.cpp:169]: No. It overrides the kill_fall method to do absolutely nothing, making the dispenser truly unkillable

Comment (MatusGuy) [reply src/badguy/dispenser.cpp:169]: wait now i get what you mean. Sure i can remove the return

**Review by tobbi — APPROVED** [review]
IDK, just looking at the code, this looks fine. Didn't test it.

**Review by Frostwithasideofsalt — APPROVED** [review]
Tested it, seems to work

**Pull request merged**